sys/suit/handlers_command_seq.c

```diff
@@ -207,34 +214,50 @@ static int _dtv_set_param(suit_manifest_t *manifest, int key,
 
     nanocbor_enter_map(it, &map);
 
+    suit_component_t *comp = _get_component(manifest);
+
     while (!nanocbor_at_end(&map)) {
         /* map points to the key of the param */
         int32_t param_key;
-        nanocbor_get_int32(&map, &param_key);
-        LOG_DEBUG("Current component index: %" PRIi32 "\n",
-                  manifest->component_current);
+        if (nanocbor_get_int32(&map, &param_key) < 0) {
+            return SUIT_ERR_INVALID_MANIFEST;
+        }
         LOG_DEBUG("param_key=%" PRIi32 "\n", param_key);
-        int res;
+        unsigned int type = nanocbor_get_type(&map);
+        /* Filter 'complex' types and only allow int, nint, bstr and tstr types
+         * for parameter values */
+        if (type > NANOCBOR_TYPE_TSTR) {
+            return SUIT_ERR_INVALID_MANIFEST;
+        }
+        suit_param_ref_t *ref;
         switch (param_key) {
-            case 6: /* SUIT URI LIST */
-                res = _param_get_uri_list(manifest, &map);
+            case SUIT_PARAMETER_VENDOR_IDENTIFIER:
+                ref = &comp->param_vendor_id;
                 break;
-            case 11: /* SUIT DIGEST */
-                res = _param_get_digest(manifest, &map);
+            case SUIT_PARAMETER_CLASS_IDENTIFIER:
+                ref = &comp->param_class_id;
                 break;
-            case 12: /* SUIT IMAGE SIZE */
-                res = _param_get_img_size(manifest, &map);
+            case SUIT_PARAMETER_IMAGE_DIGEST:
+                ref = &comp->param_digest;
+                break;
+            case SUIT_PARAMETER_COMPONENT_OFFSET:
+                ref = &comp->param_component_offset;
+                break;
+            case SUIT_PARAMETER_IMAGE_SIZE:
+                ref = &comp->param_size;
+                break;
```

Comment:
Why not set `manifest->components[manifest->component_current].size` here as well? I'm asking because currently `_print_download_progress` relies on `manifest->components[0].size` being set but it is not anymore and therefore there is no more progress bar anymore and `examples/suit_update` fails for me, any other fix with the same result works for me as well.

Reply:
Thanks for spotting. I've removed the `size` member to keep all params uniform. I've modified the coap code a bit to retrieve the max size from the manifest, similar to how it is done in other locations.